src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,58 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE_STORAGE = 10;
+    private static int sizeStorage;
+    private ElementStorage<K, V>[] storage;
+
+    public StorageImpl() {
+        this.storage = new ElementStorage[MAX_SIZE_STORAGE];
+        this.sizeStorage = 0;
+    }
+
+    class ElementStorage<K, V> {
+        private K key;
+        private V value;
+
+        public ElementStorage(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = containsKey(key);
+        if (index >= 0) {
+            storage[index] = new ElementStorage(key, value);
+        } else {
+            storage[sizeStorage++] = new ElementStorage(key, value);
+        }
+    }
+
+    private int containsKey(K key) {
```

Comment:
```suggestion
    private int getIndex(K key) {
```
private methods should go after public ones